src/otx/v2/api/entities/utils/data_utils.py

```diff
@@ -217,7 +217,18 @@ def save_image_to_cache(img: np.array, filename: str) -> None:
             if loaded_img is not None:
                 return loaded_img
 
-    img = results["dataset_item"].numpy  # this takes long for VideoFrame
+    img = results["dataset_item"].media.data  # this takes long for VideoFrame
+
+    # OTX expects RGB format
+    img = img.astype(np.uint8)
+    if len(img.shape) == 2:
+        img = cv2.cvtColor(img, cv2.COLOR_GRAY2RGB)
+    elif len(img.shape) == 3:
+        if img.shape[-1] == 3:
+            img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+        if img.shape[-1] == 4:
+            img = cv2.cvtColor(img, cv2.COLOR_BGRA2RGB)
```

Comment:
If it's not `sc_sdk.VideoFrame`, we don't need the file-based caching at all. How does the Datumaro deal with video data? The `media.data` involves on-line video frame decoding?

Reply:
With the help of workload support proxy, the video data is not expected to fed into OTX directly. More concisely, video is first decoded into video frames at sc_sdk side and these are compiled into an arrow.

But this color conversion should be more clearly inspected further, because Datumaro image is already RGB with PIL decoding.

Reply:
Happy to see the progress. @harimkang Please consider removing this video frame caching itself. It's not necessary anymore. :)